osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -126,11 +126,19 @@ public void PrepareTrackForPreviewLooping()
         }
 
         /// <summary>
-        /// Transfer a valid audio track into this working beatmap. Used as an optimisation to avoid reload / track swap
-        /// across difficulties in the same beatmap set.
+        /// Attempts to transfer the audio track to a target working beatmap, if valid for transferring.
+        /// Used as an optimisation to avoid reload / track swap across difficulties in the same beatmap set.
         /// </summary>
-        /// <param name="track">The track to transfer.</param>
-        public void TransferTrack([NotNull] Track track) => this.track = track ?? throw new ArgumentNullException(nameof(track));
+        /// <param name="target">The target working beatmap to transfer this track to.</param>
+        /// <returns>Whether the track is valid and has been transferred to this working beatmap.</returns>
+        public virtual bool TryTransferTrack([NotNull] WorkingBeatmap target)
```

Comment:
This `TransferTrack` method has been reversed in direction from inwards (transfer track from given one to ours) to outwards (transfer our track to given beatmap). 

This is necessary for `ClockBackedTestWorkingBeatmap` and `WaveformTestBeatmap` to disallow transferring their unsafe track to other working beatmaps.

We could potentially define another `ReceiveTransferredTrack` that's called on the target working beatmap, but I don't think there'll ever be a case where that's necessary.